src/compiler/checker.ts

```diff
@@ -5708,77 +5735,10 @@ namespace ts {
             }
 
             /**
-             * See signatureAssignableTo, signatureAssignableTo
+             * See signatureAssignableTo, compareSignaturesIdentical
              */
             function signatureRelatedTo(source: Signature, target: Signature, reportErrors: boolean): Ternary {

```

Comment:
signatureIsRelatedTo will have to stay inside checkTypeRelatedTo though, because it references isRelatedTo.